src/Compilers/CSharp/Portable/Symbols/Synthesized/Extensions/SynthesizedExtensionMarker.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// This method encodes the information needed to round-trip extension types
+    /// through metadata.
+    ///
+    /// It encodes:
+    /// - whether the extension type is implicit or explicit (PROTOTYPE)
+    /// - the underlying type (first parameter type)
+    /// - the base extensions (subsequent parameter types)
+    /// </summary>
```

Comment:
Can we document the metadata format here (in `<remarks>`)? Just a high-level overview that first argument is the underlying type, other arguments are base extension types. It seems this is not documented anywhere else (like in the spec).

Reply:
Added some implementation details to next draft of the spec: [https://github.com/dotnet/csharplang/pull/7038/files](https://github.com/dotnet/csharplang/pull/7038/files)
Will also improve the comment here